Add info about the Dynatrace Kubernetes Operator (#4)
dynatrace-oss-contrib/spring-boot: spring-boot-project/spring-boot-docs/src/docs/asciidoc/actuator/metrics.adoc

You can find the Dynatrace documentation on Micrometer metrics ingest {dynatrace-help}/how-to-use-dynatrace/metrics/metric-ingestion/ingestion-methods/micrometer[here].
Configuration properties in the `v1` namespace apply only when exporting to the {dynatrace-help}/dynatrace-api/environment-api/metric-v1/[Timeseries v1 API].
Configuration properties in the `v2` namespace apply only when exporting to the {dynatrace-help}/dynatrace-api/environment-api/metric-v2/post-ingest-metrics/[Metrics v2 API].
Note that this integration can export only to either the `v1` or `v2` version of the API at a time, with `v2` being the preferred one.
If the `device-id` (required for v1 but not used in v2) is set in the `v1` namespace, metrics are exported to the `v1` endpoint.
Otherwise, `v2` is assumed.

[[actuator.metrics.export.dynatrace.v2-api]]
===== v2 API

You can use the v2 API in two ways.

[[actuator.metrics.export.dynatrace.v2-api.auto-config]]
====== Auto-configuration

Dynatrace auto-configuration is available for hosts that are monitored by a OneAgent or by the Dynatrace Operator for Kubernetes.

> If a local OneAgent is running on the host, metrics are automatically exported to the {dynatrace-help}/how-to-use-dynatrace/metrics/metric-ingestion/ingestion-methods/local-api/[local OneAgent ingest endpoint].

Reviewer: Maybe we should separate these two scenarios ? Like 

## OneAgent

## Kubernetes operator

I feel it's not straightforward to identify both cases unless I read the whole thing. If it's split in sections I can quickly skip one or the other.

Author: The idea is that if you have one of the two, you dont really need to configure anything anyway, and can just start using it as is without even caring about anything else

The ingest endpoint forwards the metrics to the Dynatrace backend.

> If running in Kubernetes with the Dynatrace operator installed, the registry will pick up your endpoint URI and API token from the operator instead.

Reviewer: ```suggestion
If running in Kubernetes with the Dynatrace operator installed, the registry will automatically pick up your endpoint URI and API token from the operator instead.
```

> If running in Kubernetes with the Dynatrace operator installed, the registry will pick up your endpoint URI and API token from the operator instead.

Reviewer 2: ```suggestion
When running in Kubernetes with the Dynatrace operator installed, the registry will pick up your endpoint URI and API token from the operator instead.
```
Not sure if my suggestion here is an improvement, so feel free to ignore. :sweat_smile:

> Dynatrace operator-based autoconfiguration is available from Micrometer version `1.9.0`, which is distributed with Spring Boot since version `2.7.0`.

Reviewer 2: ```suggestion
Dynatrace Operator-based auto-configuration is available from Micrometer version `1.9.0`, which is distributed with Spring Boot since version `2.7.0`.
```

For consistency :slightly_smiling_face:

	        api-token: "YOUR_TOKEN"
----

When using the Dynatrace v2 API, the following optional features are available (more info in the {dynatrace-help}/how-to-use-dynatrace/metrics/metric-ingestion/ingestion-methods/micrometer#dt-configuration-properties[Dynatrace documentation]):

* Metric key prefix: Sets a prefix that is prepended to all exported metric keys.
* Enrich with Dynatrace metadata: If a OneAgent or Dynatrace operator is running, enrich metrics with additional metadata (for example, about the host, process, or pod).
* Default dimensions: Specify key-value pairs that are added to all exported metrics.

> If tags with the same key are specified with Micrometer, they overwrite the default dimensions.

Author: Works in the PDF and HTML preview. Unless there is something completely different happening when Spring Boot builds its documentation, I think this should work!

	        step: "30s"
----

> You can find more information on how to set up the Dynatrace exporter for Micrometer in {micrometer-registry-docs}/dynatrace[the Micrometer documentation] and {dynatrace-help}/how-to-use-dynatrace/metrics/metric-ingestion/ingestion-methods/micrometer[the Dynatrace documentation].

Reviewer 3: ```suggestion
You can find more information on how to set up the Dynatrace exporter for Micrometer in the {micrometer-registry-docs}/dynatrace[Micrometer documentation] and the {dynatrace-help}/how-to-use-dynatrace/metrics/metric-ingestion/ingestion-methods/micrometer[Dynatrace documentation].
```
